roles/ovirt-datacenter-cleanup/tasks/main.yml

```diff
@@ -15,22 +15,33 @@
     auth: "{{ ovirt_auth }}"
     pattern: "datacenter={{ data_center_name }}"
 
-- name: Remove all Storage Domains except master
+- name: Remove all Storage Domains except master (and hosted_storage if HE)
   include: storages_pre.yml
 
 - name: Find existing clusters in Datacenter
-  ovirt_clusters_facts:
+  ovirt_cluster_facts:
     auth: "{{ ovirt_auth }}"
     pattern: "datacenter={{ data_center_name }}"
 
 - name: Remove Datacenter
   include: datacenter.yml
+  when:
+    - not hosted_engine
```

Comment:
when you have only one condition, IMHO it will be nice to do it in the same line like:
when: not hosted_engine

http://docs.ansible.com/ansible/latest/playbooks_conditionals.html#the-when-statement
they use it in the example as you did just in case of Multiple conditions that all need to be true (a logical ‘and’) can also be specified as a list